drivers/sensor/adt7310/Kconfig

```diff
@@ -0,0 +1,61 @@
+# ADT7310 temperature sensor configuration options
+
+# Copyright (c) 2023 Andriy Gelman
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig ADT7310
+	bool "ADT7310 Temperature Sensor"
+	default y
+	depends on DT_HAS_ADI_ADT7310_ENABLED
+	select SPI
+	help
+	  Enable the driver for Analog Devices ADT7310 High-Accuracy
+	  16-bit Digital SPI Temperature Sensors.
+
+if ADT7310
+
+config ADT7310_TRIGGER
+	bool "Trigger interrupt during over/under temperature"
+	depends on GPIO
```

Comment:
Make this a promptless config, then have `ADT7310_TRIGGER_GLOBAL_THREAD` and`ADT7310_TRIGGER_OWN_THREAD` select it